feat: add C implementation for `stats/base/dists/uniform/mean` (#3964)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/uniform/mean/README.md

@license Apache-2.0

> Copyright (c) 2024 The Stdlib Authors.

Reviewer: ```suggestion
Copyright (c) 2018 The Stdlib Authors.
```

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/uniform/mean.h"
```

#### stdlib_base_dists_uniform_mean( a, b )

> Returns the [mean][mean] of a [uniform][uniform-distribution] distribution with minimum value \( a \) and maximum value \( b \).

Reviewer: ```suggestion
Returns the [expected value][expected-value] of a [uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).
```

```c
double out = stdlib_base_dists_uniform_mean( 0.0, 1.0 );
// returns 0.5
```

The function accepts the following arguments:

> -   **a**: `[in] double` minimum value.
> -   **b**: `[in] double` maximum value.

Reviewer: ```suggestion
-   **a**: `[in] double` minimum support.
-   **b**: `[in] double` maximum support.
```

```c
double stdlib_base_dists_uniform_mean( const double a, const double b );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/uniform/mean.h"
#include <stdlib.h>
#include <stdio.h>

> 

Reviewer: You mind adding

```
static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}
```

here and using it to draw the random numbers here, please? Applies to the example and benchmark C files, too.

int main( void ) {
    double a;
    double b;
    double y;
    int i;

    for ( i = 0; i < 25; i++ ) {
        a = (double)( rand() % 10 );
        b = a + (double)( rand() % 10 + 1 ); // ensure b > a
        y = stdlib_base_dists_uniform_mean( a, b );

>         printf( "a: %lf, b: %lf, Mean(X;a,b): %lf\n", a, b, y );

Reviewer: ```suggestion
        printf( "a: %lf, b: %lf, E(X;a,b): %lf\n", a, b, y );
```

> int main( void ) {
>     double a;
>     double b;
>     double y;
>     int i;
> 
>     for ( i = 0; i < 25; i++ ) {
>         a = (double)( rand() % 10 );
>         b = a + (double)( rand() % 10 + 1 ); // ensure b > a
>         y = stdlib_base_dists_uniform_mean( a, b );
>         printf( "a: %lf, b: %lf, E(X;a,b): %lf\n", a, b, y );
>     }
> }

Reviewer: ```suggestion
static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double a;
    double b;
    double y;
    int i;

    for ( i = 0; i < 25; i++ ) {
        a = random_uniform( 0.0, 10.0 );
        b = random_uniform( a, 20.0 ); // ensure b > a
        y = stdlib_base_dists_uniform_mean( a, b );
        printf( "a: %lf, b: %lf, E(X;a,b): %lf\n", a, b, y );
    }
}
```

[uniform-distribution]: https://en.wikipedia.org/wiki/Uniform_distribution_%28continuous%29

> [mean]: https://en.wikipedia.org/wiki/Mean
> 

Reviewer: ```suggestion
```
